### ISSUE TITLE
Debian: Cannot build wallet from source

### ISSUE THREAD

**Issue by hartrock** [iotaledger/trinity-wallet#686]
### Bug description
Wallet doesn't run correctly under Debian Linux. After starting it only empty window showing background appears.

Trying to build myself gives errors.

+++
sr@free:~/Projects/trinity-wallet-0.4.3$ npm run deps:desktop

> trinity-wallet@0.0.0 deps:desktop /home/sr/Projects/trinity-wallet-0.4.3
> cd src/desktop && npm install && cd ../..

> usb@1.3.3 install /home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download 

node-pre-gyp WARN Tried to download(404): https://github.com/tessel/node-usb/releases/download/1.3.3/usb_bindings-v1.3.3-node-v67-linux-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for usb@1.3.3 and node@11.3.0 (node-v67 ABI, glibc) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Pre-built binaries not installable for usb@1.3.3 and node@11.3.0 (node-v67 ABI, glibc) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error Connection closed while downloading tarball file 
make: Entering directory '/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/build'
  CC(target) Release/obj.target/libusb/libusb/libusb/core.o
make: Entering directory '/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/build'
  CC(target) Release/obj.target/libusb/libusb/libusb/core.o
rm: cannot remove './Release/.deps/Release/obj.target/libusb/libusb/libusb/core.o.d.raw': No such file or directory
libusb.target.mk:140: recipe for target 'Release/obj.target/libusb/libusb/libusb/core.o' failed
make: *** [Release/obj.target/libusb/libusb/libusb/core.o] Error 1
make: Leaving directory '/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/build'
  CC(target) Release/obj.target/libusb/libusb/libusb/descriptor.o
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
gyp ERR! System Linux 4.9.0-8-amd64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding/usb_bindings.node" "--module_name=usb_bindings" "--module_path=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding" "--napi_version=3" "--node_abi_napi=napi"
gyp ERR! cwd /home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb
gyp ERR! node -v v11.3.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding/usb_bindings.node --module_name=usb_bindings --module_path=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding --napi_version=3 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:978:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
node-pre-gyp ERR! System Linux 4.9.0-8-amd64
node-pre-gyp ERR! command "/usr/bin/node" "/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb
node-pre-gyp ERR! node -v v11.3.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.3
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding/usb_bindings.node --module_name=usb_bindings --module_path=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding --napi_version=3 --node_abi_napi=napi' (1)
../libusb/libusb/descriptor.c:1199:1: fatal error: opening dependency file ./Release/.deps/Release/obj.target/libusb/libusb/libusb/descriptor.o.d.raw: No such file or directory
 }
 ^
compilation terminated.
libusb.target.mk:140: recipe for target 'Release/obj.target/libusb/libusb/libusb/descriptor.o' failed
make: *** [Release/obj.target/libusb/libusb/libusb/descriptor.o] Error 1
make: Leaving directory '/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
gyp ERR! System Linux 4.9.0-8-amd64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding/usb_bindings.node" "--module_name=usb_bindings" "--module_path=/home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb/src/binding" "--napi_version=3" "--node_abi_napi=napi"
gyp ERR! cwd /home/sr/Projects/trinity-wallet-0.4.3/src/desktop/node_modules/usb
gyp ERR! node -v v11.3.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.0.17 (node_modules/kdbxweb/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! usb@1.3.3 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the usb@1.3.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sr/.npm/_logs/2018-12-02T23_55_20_705Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! trinity-wallet@0.0.0 deps:desktop: `cd src/desktop && npm install && cd ../..`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the trinity-wallet@0.0.0 deps:desktop script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sr/.npm/_logs/2018-12-02T23_55_20_728Z-debug.log
+++

### Environment Specs
What device are you running Trinity on?
ThinkPad.
- OS version:
sr@free:~/Projects/trinity-wallet-0.4.3$ uname -a
Linux free 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux

What build of Trinity are you running? Please provide the full version (you can find it in Settings > About Trinity).
- Trinity version: 
trinity-wallet-0.4.3

### Steps To Reproduce
1. See above.

### Expected behaviour
No error.

### Actual behaviour
Error.

### Errors
See above.

**Comment by rajivshah3** [comment]
@hartrock can you install `node` 8 or 9 and try again? I think I saw similar issues when trying to compile from source with `node` 10 or 11

Edit: According to `node-usb`'s readme, you may need to also do `sudo apt-get install build-essential libudev-dev` to build from source (https://github.com/tessel/node-usb#installation)

**Comment by rajivshah3** [comment]
Possibly a duplicate of #534

**Comment by hartrock** [comment]
> apt-get install build-essential libudev-dev

**Comment by hartrock** [comment]
> Edit: According to `node-usb`'s readme, you may need to also do `sudo apt-get install build-essential libudev-dev` to build from source (https://github.com/tessel/node-usb#installation)

This has not been a problem (already installed).

**Comment by hartrock** [comment]
> @hartrock can you install `node` 8 or 9 and try again? I think I saw similar issues when trying to compile from source with `node` 10 or 11

Downgrading lacks npm (even conflicts):
```
sr@free:~/tmp$ aptitude show nodejs
Package: nodejs                          
Version: 11.3.0-1nodesource1
State: installed (4.8.2~dfsg-1), upgrade available (11.3.0-1nodesource1)
Automatically installed: no
Priority: optional
Section: web
Maintainer: Chris Lea <chl@nodesource.com>
Architecture: amd64
Uncompressed Size: 73.1 M
Depends: libc6 (>= 2.9), libgcc1 (>= 1:3.4), libstdc++6 (>= 4.4.0),
         python-minimal, ca-certificates
Conflicts: nodejs-dev, nodejs-legacy, npm
Replaces: nodejs-dev (<= 0.8.22), nodejs-legacy, npm (<= 1.2.14)
Provides: nodejs-dev, nodejs-legacy, npm
Description: Node.js event-based server-side javascript engine
 Node.js is similar in design to and influenced by systems like Ruby's Event
 Machine or Python's Twisted. 
 
 It takes the event model a bit further - it presents the event loop as a
 language construct instead of as a library. 
 
 Node.js is bundled with several useful libraries to handle server tasks :
 System, Events, Standard I/O, Modules, Timers, Child Processes, POSIX, HTTP,
 Multipart Parsing, TCP, DNS, Assert, Path, URL, Query Strings.
Homepage: https://nodejs.org
Tags: devel::interpreter, devel::lang:ecmascript, devel::runtime, devel::web,
      implemented-in::c++, implemented-in::ecmascript, interface::commandline,
      interface::shell, protocol::dns, protocol::http, protocol::tcp,
      role::program, scope::application, works-with-format::json
```
Current version (with build problem) is:
```
sr@free:~/tmp$ aptitude show nodejs
Package: nodejs                          
Version: 11.3.0-1nodesource1
State: installed (4.8.2~dfsg-1), upgrade available (11.3.0-1nodesource1)
Automatically installed: no
Priority: optional
Section: web
Maintainer: Chris Lea <chl@nodesource.com>
Architecture: amd64
Uncompressed Size: 73.1 M
Depends: libc6 (>= 2.9), libgcc1 (>= 1:3.4), libstdc++6 (>= 4.4.0),
         python-minimal, ca-certificates
Conflicts: nodejs-dev, nodejs-legacy, npm
Replaces: nodejs-dev (<= 0.8.22), nodejs-legacy, npm (<= 1.2.14)
Provides: nodejs-dev, nodejs-legacy, npm
Description: Node.js event-based server-side javascript engine
 Node.js is similar in design to and influenced by systems like Ruby's Event
 Machine or Python's Twisted. 
 
 It takes the event model a bit further - it presents the event loop as a
 language construct instead of as a library. 
 
 Node.js is bundled with several useful libraries to handle server tasks :
 System, Events, Standard I/O, Modules, Timers, Child Processes, POSIX, HTTP,
 Multipart Parsing, TCP, DNS, Assert, Path, URL, Query Strings.
Homepage: https://nodejs.org
Tags: devel::interpreter, devel::lang:ecmascript, devel::runtime, devel::web,
      implemented-in::c++, implemented-in::ecmascript, interface::commandline,
      interface::shell, protocol::dns, protocol::http, protocol::tcp,
      role::program, scope::application, works-with-format::json
```
Debian version:
```
sr@free:~$ uname -a
Linux free 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux
sr@free:~$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by hartrock** [comment]
Current try:
```
sr@free:~/Projects/trinity-wallet$ npm run deps:desktop > /tmp/log.txt 2>&1 
```
(log attached).
[log.txt](https://github.com/iotaledger/trinity-wallet/files/2646386/log.txt)

**Comment by rajivshah3** [comment]
Closed in #779
